tools/cli/commands/create.py

```diff
@@ -0,0 +1,375 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Methods for implementing the `datalab create` command."""
+
+import argparse
+import subprocess
+import tempfile
+
+import connect
+
+
+description=(
+"""{0} {1} creates a new Datalab instances running in a Google
+Compute Engine VM.
+
+This command also creates the 'datalab-network' network if necessary.
+
+By default, the command creates a persistent connection to the newly
+created instance. You can disable that behavior by passing in the
+'--no-connect' flag."""
+)
+
+
+_DATALAB_NETWORK = 'datalab-network'
+_DATALAB_NETWORK_DESCRIPTION = 'Network for Google Cloud Datalab instances'
+
+_DATALAB_FIREWALL_RULE = 'datalab-network-allow-ssh'
+_DATALAB_FIREWALL_RULE_DESCRIPTION = 'Allow SSH access to Datalab instances'
+
+_DATALAB_DEFAULT_DISK_SIZE_GB = 200
+_DATALAB_DISK_DESCRIPTION = (
+    'Persistent disk for a Google Cloud Datalab instance')
+
+_DATALAB_STARTUP_SCRIPT = """#!/bin/bash
```

Comment:
Does it make sense to write this script file as a .sh and package it as a file in the tool, rather than embed in the code.

[not for now] but think this is an interesting user customization point - to be able to provide a modified startup script for the VM.